### PR TITLE
Cleanup public APIs, constants, and helper class (resolves #4)

### LINKED ISSUE #4: public APIs, helper classes and constants
- [x] Rename package to com.adobe.marketing.mobile.edge.media
- [ ] No changes should be surfaced in the public API/constant names
- [ ] Check iOS if the keys for Media.create* APIs have been updated to match the new XDM like keys

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

More cleanup of public APIs, constants, and helper classes.

- Move `MediaTrackerEventGenerator` class to "internal" package and remove duplicate constants to use `MediaInternalConstants` instead.
- Rename `MediaTrackerInterface` to `MediaEventTracking` to match iOS implementation.
- Update event source and event type constants.  Temporarily using these constants until EventType and EventSource in Core is updated.
- Fix Java annotations to use AndroidX lib

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java [hunk 1]
```diff
@@ -20,83 +29,21 @@
...
-            static final String ID = "error.id";
-        }
-    }
+public class MediaTrackerEventGenerator implements MediaTracker {
+    private static final String SOURCE_TAG = "MediaTrackerEventGenerator";
 
     private static final int TICK_INTERVAL_MS = 750;
     private static final int EVENT_TIMEOUT_MS = 500;
     private final AdobeCallback<Event> eventConsumer;
-    private final Map<String, Object> config;
```
Comment (kevinlind) [inline code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:87]: This `config` variable wasn't used so I removed it from the constructor.

File: code/edgemedia/src/test/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionTrackerTests.java [hunk 2]
```diff
@@ -114,8 +113,7 @@ public MediaCollectionTrackerTests() {
 
         Map<String, Object> config = new HashMap<>();
 
-        mediaTrackerAPIEventGenertor =
-                TestMediaTrackerEventGenerator.create(config, "tracker0", true);
+        mediaTrackerAPIEventGenertor = TestMediaTrackerEventGenerator.create("tracker0", true);
```
Comment (addb) [inline code/edgemedia/src/test/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionTrackerTests.java:116]: Let's rename to `MediaPublicTracker` similar to iOS

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java [hunk 3]
```diff
@@ -20,83 +29,21 @@
...
-            static final String STATE_END = "stateend";
-        }
-
-        static final class ErrorInfo {
-            private ErrorInfo() {}
-
-            static final String ID = "error.id";
-        }
-    }
+public class MediaTrackerEventGenerator implements MediaTracker {
```
Comment (addb) [inline code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:32]: Let's rename this class to `MediaPublicTracker` similar to iOS

File: code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionTracker.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-class MediaCollectionTracker implements MediaTrackerInterface {
+class MediaCollectionTracker implements MediaEventTracking {
```
Comment (addb) [inline code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionTracker.java:26]: Let's rename this class to `MediaEventTracker ` similar to iOS

File: code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionConstants.java [hunk 5]
```diff
@@ -16,7 +16,7 @@ class MediaCollectionConstants {
     private MediaCollectionConstants() {}
 
     static final String MEDIA_LIBRARY_VERSION =
-            "android-media-" + com.adobe.marketing.mobile.Media.extensionVersion();
+            "android-media-" + com.adobe.marketing.mobile.edge.media.Media.extensionVersion();
```
Comment (addb) [inline code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionConstants.java:19]: We can remove this file entirely.

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java (hunk 3 above)
Comment (kevinlind) [reply code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:32]: I'll leave that for future work in another PR when we add the MediaSession and MediaEventTracker classes.

File: code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionTracker.java (hunk 4 above)
Comment (kevinlind) [reply code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionTracker.java:26]: I'll leave that for future work in another PR when we add the MediaSession and MediaEventTracker classes.

File: code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionConstants.java (hunk 5 above)
Comment (kevinlind) [reply code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaCollectionConstants.java:19]: I'll leave that for future work in another PR when we add the MediaSession and MediaEventTracker classes.

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java (hunk 3 above)
Comment (kevinlind) [reply code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java]: On second thought, for this one I will rename to MediaPublicTracker, as there won't be much more work on it.

**Review by addb — APPROVED** [review]

**Pull request merged**